### PR TITLE
Lib: fix #333

### LINKED ISSUE #333: `Tyxml_js.R.filter_attrib`
@hhugo Can you implement this one in jsoo, similar to the one you did in eliom ? 

I wonder what it would cost us to have that directly in tyxml ...

### PR DESCRIPTION
**Pull request by hhugo**
@Drup

### CODE REVIEW

File: lib/tyxml/tyxml_js.ml [hunk 1]
```diff
@@ -42,13 +42,10 @@ module Xml = struct
   type keyboard_event_handler = Dom_html.keyboardEvent Js.t -> bool
   type attrib_k =
     | Event of biggest_event_handler
-    | Attr of Dom.attr Js.t
+    | Attr of Js.js_string Js.t option React.S.t
```
Comment (Drup) [inline lib/tyxml/tyxml_js.ml]: Would it make sense (from an efficiency perspective) to keep a constant attribute constructor, with the new signal attribute ?

File: lib/tyxml/tyxml_js.ml [hunk 2]
```diff
@@ -78,9 +75,14 @@ module Xml = struct
 
   let attach_attribs e l =
     List.iter (fun (n,att) ->
+      let n = Js.string n in
         match att with
-        | Attr a -> ignore(e##setAttributeNode(a))
-        | Event h -> Js.Unsafe.set e (Js.string n) (fun ev -> Js.bool (h ev))
+        | Attr a ->
+          let _ : unit React.S.t = React.S.map (function
```
Comment (Drup) [inline lib/tyxml/tyxml_js.ml]: Assuming that GC of signals on javascript might be one day fixed, shouldn't this use `Lwt_react.S.keep` ?

**Comment by Drup** [comment]
Great, thanks!

**Comment by Drup** [comment]
cc @slegrand45

File: lib/tyxml/tyxml_js.ml (hunk 1 above)
Comment (hhugo) [inline lib/tyxml/tyxml_js.ml]: React already has fast path for constant signals. I don't think you'll gain any efficiency adding an extra constructor.

File: lib/tyxml/tyxml_js.ml (hunk 2 above)
Comment (hhugo) [inline lib/tyxml/tyxml_js.ml]: I'll just add a comment about it.

File: lib/tyxml/tyxml_js.ml (hunk 1 above)
Comment (Drup) [inline lib/tyxml/tyxml_js.ml]: Ok.

**Pull request merged**